core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/provider.go

```diff
@@ -131,16 +135,31 @@ func (p *logEventProvider) SetConfig(cfg ocr2keepers.LogEventProviderConfig) {
 	logLimit := cfg.LogLimit
 
 	if blockRate == 0 {
-		blockRate = defaultBlockRateForChain(p.chainID)
+		blockRate = p.opts.defaultBlockRate()
 	}
 	if logLimit == 0 {
-		logLimit = defaultLogLimitForChain(p.chainID)
+		logLimit = p.opts.defaultLogLimit()
 	}
 
 	p.lggr.With("where", "setConfig").Infow("setting config ", "bockRate", blockRate, "logLimit", logLimit)
 
-	// TODO set block rate and log limit on the buffer
-	//p.buffer.SetConfig(blockRate, logLimit)
+	atomic.StoreUint32(&p.opts.BlockRate, blockRate)
+	atomic.StoreUint32(&p.opts.LogLimit, logLimit)
```

Comment:
nit. in function below to set buffer version we use p.lock to manage concurrency, might be better to use that in this function too for uniformity and it will automatically apply to all config changed in this function

Reply:
It would be good in general to acquire a lock while updating config